calendar/calendar-tool/tool/src/webapp/js/sakai-calendar.js

```diff
@@ -1,110 +1,155 @@
+                icon: event.eventIcon,
+                event_id:  event.eventId,
+                attachments: event.attachments,
+                eventReference: event.eventReference,
+                end: endDate
+              });
             });
-          });
-          successCallback(events);
-        }).catch(error => console.error(error));
+            successCallback(events);
+          }).catch(error => console.error(error));
+        }
+      }],
+      eventSourceSuccess: function(content, xhr) {
+        return content.eventArray;
+      },
+      eventBackgroundColor: eventBackgroundColor,
+      eventTextColor: eventTextColor,
+      eventContent: function (args) {
+        const isListView = SakaiCalendar.calendar.getCurrentData() && SakaiCalendar.calendar.getCurrentData().currentViewType.includes('list');
+        // Create the default event structure to benefit from the theme.
+        const eventTitle = document.createElement('div');
+        eventTitle.classList.add(!isListView ? 'fc-event-title' : 'fc-list-event-title');
+        // Build our event HTML including the default icon for the event type.
+        const icon = args.event._def.extendedProps.icon;
+        const title = args.event._def.title;
+        const titleElement = document.createTextNode(title);
+        const type = args.event._def.extendedProps.type;
+        const eventUrl = args.event._def.url;
+        const eventLink = document.createElement('a');
+        eventLink.href = eventUrl;
+        eventLink.title = title;
+        eventLink.appendChild(titleElement);
+        const iconSpan = document.createElement('span');
+        iconSpan.classList.add('icon');
+        iconSpan.classList.add(icon);
+        iconSpan.title = type;
+        // Append our custom event with icon to the default event structure.
+        eventTitle.appendChild(iconSpan);
+        eventTitle.appendChild(!isListView ? titleElement : eventLink);
+        // Return the modified event including out event structure.
+        return {
+          html: eventTitle.outerHTML
+        };
+      }
+    });
+
+    userLanguage && this.calendar.setOption('locale', userLanguage);
+    this.calendar.render();
+  },
+
+  // Go to the current date set by the backend.
+  gotoDate: function(date) {
+    this.calendar.gotoDate(currentDate);
+  },
+
+  // This logic is associated to set the default subview, by day, month, week or list.
+  setDefaultSubview: function(defaultSubview) {
+    switch (defaultSubview) {
+      case 'day':
+        this.calendar.changeView('timeGridDay');
+        break;
+      case 'month':
+        this.calendar.changeView('dayGridMonth');
+        break;
+      case 'list':
+        this.calendar.changeView('listWeek');
+        break;
+      case 'week':
+      default:
+        this.calendar.changeView('timeGridWeek');
+  }
+
+    // When the user changes the view, reflect the change in a param to set the default view.
+    setDefaultCalendarSubview = () => {
```

Comment:
I think you can add this to the namespace, too. Just reference it in your onclick handler further down.

Reply:
Okay